### PR TITLE
Add support for line plotting

### PR DESCRIPTION
**Pull request by beattie**
I've never used the github pull request, so I hope I have this right.  If not, or you would rather get this via a different method, just let me know.

There are two commits, one adds support for generating a line plot.  The other one adds a test program to test line plotting and uses a glade file to create the window.

Since I don't know waf I created a Makefile,  I'll look into using waf to build the new test program.

### CODE REVIEW

File: tests/Makefile [hunk 1]
```diff
@@ -0,0 +1,6 @@
+LIBS=$(shell pkg-config --libs gtk+-3.0) -lm ../build/libgoatplot.so
```
Comment (drahnr) [inline tests/Makefile]: Please just add another test program entry to the wscript in the root of this repository.
Please rename the the test to glade-line.c
```
	test_glade = bld.program(
		features = ['c', 'glib2', 'unites'],
		target = 'test-glade-line',
		source = ['tests/glade-line.c'],
		includes = ['src/'],
		export_includes = ['src/'],
		use = 'objects',
		uselib = 'M GOBJECT GLIB GTK3',
		install_path = None
	)

```

File: tests/gladeTest.c [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+	Both both;
+
+	gtk_init(&argc, &argv);
+
+	plot = goat_plot_new();
+
+	if((builder = gtk_builder_new_from_file("test.glade")) ==
+			NULL) {
+		fprintf(stderr, "gtk_builder_new failed\n");
+		exit(-1);
```
Comment (drahnr) [inline tests/gladeTest.c]: https://developer.gnome.org/glib/stable/glib-Message-Logging.html#g-error
`
g_error(...)` is a shorter and cleaner way to do so

File: tests/gladeTest.c [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+	}
+
+	if((window = GTK_WIDGET(gtk_builder_get_object(builder, "MainWindow"))) ==
+			NULL) {
+		fprintf(stderr, "gtk_builder_get_object failed\n");
+		exit(-1);
```
Comment (drahnr) [inline tests/gladeTest.c]: see above

File: tests/gladeTest.c [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+	};
+
+	if((plot = GOAT_PLOT(gtk_builder_get_object(builder, "GoatPlot"))) ==
+			NULL) {
+		fprintf(stderr, "gtk_builder_get_object GoatPlot failed\n");
+		exit (-1);
```
Comment (drahnr) [inline tests/gladeTest.c]: see above

File: tests/test.glade [hunk 5]
```diff
@@ -0,0 +1,42 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (drahnr) [inline tests/test.glade]: please rename to match your test name :) other than that, :+1:

File: src/goat-plot.c [hunk 6]
```diff
@@ -402,7 +413,11 @@ draw_dataset (GoatPlot *plot, cairo_t *cr,
 			return FALSE;
 		}
 	}
-	cairo_fill (cr);
+	if(cairo_draw_filled) {
+		cairo_fill (cr);
+	} else {
+		cairo_stroke (cr);
+	}
```
Comment (drahnr) [inline src/goat-plot.c]: looks good :+1:

**Comment by drahnr** [comment]
The PR itself is fine. Do not get discouraged, the above are just a few minor things.

Thanks for the PR :+1:

**Comment by beattie** [comment]
I'll fix that stuff.  I need to update waf to compile the test before I add it to the tests.

**Comment by beattie** [comment]
Added your lines to wscript, however glade-line does not build

**Pull request merged**